src/main/java/com/bakdata/kafka/KafkaStreamsApplication.java

```diff
@@ -277,9 +269,14 @@ protected Properties createKafkaProperties() {
         return kafkaConfig;
     }
 
+    /**
+     * Run the Streams application. This method blocks until Kafka Streams has completed shutdown, either because it
+     * caught an error or the application has received a shutdown event.
+     */
     protected void runStreamsApplication() {
         this.streams.start();
         Runtime.getRuntime().addShutdownHook(new Thread(this::close));
+        await().forever().until(this::hasStreamsShutdown);
```

Comment:
I think you can solve this by creating a `new CountDownLatch(1)`, counting it down inside the shutdown hook thread, and awaiting it here.

Reply:
Also, maybe we could expose the awaiting as a separate method? This way we can overwrite `runStreamsApplication` if we have to set up stuff after starting Kafka Streams.

Reply:
What do you think now?